File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {

    public List<TbBrand> findAll();

    /**
     * 分页查询品牌列表
     * @param pageNum 当前页
     * @param pageSize 每页显示条数
     * @return 结果集
     */
    public PageResult findPage(int pageNum, int pageSize);

    /**
     * 添加品牌列表
     * @param tbBrand
     */
    public void save(TbBrand tbBrand);

    /**
     * 根据ID查询单个品牌
     * @param id
     */
    public TbBrand selectOne(long id);

    /**
     * 修改品牌
     * @param tbBrand
     */
    public void update(TbBrand tbBrand);

    /**
     * 删除品牌
     * @param ids
     */
    public void delete(String[] ids);

    /**
     * 按条件查询品牌列表,方法的重载
     * @param tbBrand 封装查询条件的实体
     * @param pageNum 当前页
     * @param pageSize 每页显示条数
     * @return 返回查询结果
     */
    public PageResult findPage(TbBrand tbBrand, int pageNum, int pageSize);

    public List<Map<String, Object>> selectAll();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js


//自定义服务,参数1:服务名称,参数2:服务中的内容.用到其他服务就注入
//所谓服务就是处理数据的,与后台进行交互的
//服务的提取与后台交互的部分进行提取

app.service("brandService",function ($http) {
    this.findAll=function () {
        return $http.get("../brand/findAll.do");
    }

    this.findPage=function (pageNum,pageSize) {
        return $http.get("../brand/findPage.do?pageNum="+pageNum+"&pageSize="+pageSize);
    }

    this.search=function (pageNum,pageSize,searchEntity) {
        return $http.post("../brand/search.do?pageNum="+pageNum+"&pageSize="+pageSize,searchEntity);
    }

    this.saveOrUpdate=function (entity) {
        return $http.post("../brand/saveOrUpdate.do",entity);
    }

    this.selectOne=function (id) {
        return $http.get("../brand/selectOne.do?id="+id);
    }

    this.deleteChecked=function (ids) {
        return $http.post("../brand/delete.do",ids);
    }

    this.selectAll=function () {
        return $http.get("../brand/selectAll.do");
    }

});